var/spack/repos/builtin/packages/mochi-thallium/package.py

```diff
@@ -52,7 +53,8 @@ class MochiThallium(CMakePackage):
     depends_on('mochi-margo@0.6:', when='@0.5:')
     depends_on('mochi-margo@0.5:', when='@0.4:0.4.2')
     depends_on('mochi-margo@0.4:', when='@:0.3.4')
-    depends_on('cereal', when='+cereal')
+    depends_on('cereal@:1.3.0', when='@0.4.1:0.10.0 +cereal')
+    depends_on('cereal@1.3.1:', when='@0.10.1: +cereal')
```

Comment:
FYI, you can group common requirements together with the `when` context manager (in case you find it more maintainable or more readable):
```suggestion
    with when('+cereal'):
        depends_on('cereal@:1.3.0', when='@0.4.1:0.10.0')
        depends_on('cereal@1.3.1:', when='@0.10.1:')
```